src/pip/_internal/wheel.py

```diff
@@ -204,6 +204,13 @@ def message_about_scripts_not_on_PATH(scripts):
     return "\n".join(msg_lines)
 
 
+# We need this to avoid TypeError because values at a given tuple index
+# can be strings in some rows and integers in others.
+def sorted_outrows(outrows):
+    """Return the given "outrows" in sorted order."""
+    return sorted(outrows, key=lambda row: tuple(str(x) for x in row))
```

Comment:
Would it be possible to coerce everything to string when `outrows` is appended to, instead to needing to deal with mixed types?

Reply:
There was an interesting discussion at the original issue after I wrote this PR:
https://github.com/pypa/pip/issues/5868
So I think I actually want to "withdraw" this now. :) Or at least rethink it first as I think some decisions need to be made. It might be better to discuss at that issue.

I may close this or mark as "WIP" in the meantime.

Reply:
@uranusjr In thinking more about this, I'm starting to think that what I originally proposed is okay. There are two reasons: (1) Coercing everything to a string on append seems more brittle because you need to add that logic each place you are appending, which can be multiple spots (or remember to use a common helper function when appending). (2) Coercing everything to a string seems to violate the spirit of PEP 376. That PEP says the third element should be a size (i.e. integer). Thus I think it would be better / safer to leave the rows themselves alone, and confine the coercion to the sort operation (which is just a cosmetic thing anyways).

Reply:
I think sorting on the file path should be enough ? We should not be getting two lines for the same file.
(And add a warning/error if we end up with duplicate lines)

Reply:
I think sorting on the file path should be enough ? We should not be getting two lines for the same file.
(And add a warning/error if we end up with duplicate lines)

Reply:
I think sorting on the file path should be enough ? We should not be getting two lines for the same file.
(And add a warning/error if we end up with duplicate lines)

Reply:
I think sorting on the file path should be enough ? We should not be getting two lines for the same file.
(And add a warning/error if we end up with duplicate lines)

Reply:
I think sorting on the file path should be enough ? We should not be getting two lines for the same file.
(And add a warning/error if we end up with duplicate lines)

Reply:
Thanks, @xavfernandez. I definitely support at least adding a warning, but I think that should be done as part of a separate issue and PR so as not to expand the scope. I meant for this PR only to prevent the sort operation from crashing. 

Re: sorting by only the first element, it's true that using all elements might almost never matter, but is there any harm? Being able to guarantee determinism even in unlikely edge cases or error cases seems like a good thing. If we add validation later to ensure the file names will be unique, we can always adjust the sort operation then.

Reply:
It's fine I guess, maybe with a comment explaining the expected format (name, hash, size) and the fact that we are ok with sorting integer as string (since normally the sorting only happens on name)

Reply:
The current implementation + a comment explaining nuances (maybe with a pointer to this issue and/or #5868) should be enough IMO.

Reply:
Sounds like a good solution, @xavfernandez and @uranusjr. Thanks. I'll draft up a comment and repost.